### PR TITLE
Add support for including headers when importing XCFrameworks assembled with static libraries

### PR DESCRIPTION
**Pull request by thii**
For example, with the following XCFramework:

```
ios_static_xcframework.xcframework
├── Info.plist
├── ios-arm64
│   ├── Headers
│   │   └── shared.h
│   └── ios_static_xcframework.apple_static_library_lipo.a
└── ios-arm64_x86_64-simulator
    ├── Headers
    │   └── shared.h
    └── ios_static_xcframework.apple_static_library_lipo.a
```

Specifying `includes = ["Headers"]` will pass `-Ipath/to/ios_static_xcframework.xcframework/ios-arm64_x86_64-simulator/Headers` to the `{cc,objc}_library` that depends on this, when building for the iOS Simulator platform.

### CODE REVIEW

**Review by keith — APPROVED** [review]

**Review by brentleyjones — APPROVED** [review]

File: apple/internal/apple_framework_import.bzl [hunk 1]
```diff
@@ -616,8 +617,16 @@ def _common_static_framework_import_impl(ctx, is_xcframework):
     providers.append(
         _objc_provider_with_dependencies(ctx, objc_provider_fields, additional_objc_infos),
     )
+
+    includes = []
+    if is_xcframework and not is_framework and ctx.attr.includes:
```
Comment (rockwotj) [inline apple/internal/apple_framework_import.bzl:622]: Why is there a `not is_framework` here?

Comment (keith) [reply apple/internal/apple_framework_import.bzl:622]: Note that this has changed slightly at HEAD

Comment (rockwotj) [reply apple/internal/apple_framework_import.bzl:622]: It looks to be broken at HEAD? It doesn't look like there is a prefix added but I'm not exactly sure where those are supposed to be relative too: https://cs.github.com/bazelbuild/rules_apple/blob/76752fe84ff0dc5d793033015647c78efbf41309/apple/internal/apple_xcframework_import.bzl#L578

Comment (keith) [reply apple/internal/apple_framework_import.bzl:622]: do you have a simple case I can test with? definitely possible something regressed

**Pull request merged**